testSuite/__main__.py

```diff
@@ -38,8 +38,7 @@ class Test(unittest.TestCase):
             @classmethod
             def setUpClass(cls):
                 print('Running test: %s' % name)
-                args = ['python3', '-m', self._compiler, '-agfjs', '-O', tmpDir, testFileName]
-                print(' '.join(args))
+                args = ['python', '-m', self._compiler, '-agfjs', '-O', tmpDir, testFileName]

```

Comment:
on the systems with python2 (the most of them) `python` is usually python 2. We have dropped python 3, this should be tested against python 3. In fact we need argv[0] in C sense instead of hardcoded string. I don't know a cross-platform cross-implementation way to get argv[0] in C sense. I know the way to get it for cpython, but for the rest of pythons it won't work. For posix OSes we can get current process pid, find the processes with that pid and parse their command line. Anyway, the library to do so (it is possible to write a one) will be very hacky and pretty unreliable.

Reply:
for local testing I suggest using tox. tox is a generic virtual control and test command line tool.

Reply:
No tox , nose and pytest, please.